cashu/wallet/wallet.py

```diff
@@ -8,6 +8,7 @@
 from datetime import datetime, timedelta
 from itertools import groupby
 from typing import Dict, List, Optional, Tuple, Union
+from urllib.parse import urljoin

```

Comment:
`urljoin` can behave a bit weird:

```python
In [1]: from urllib.parse import urljoin

In [2]: urljoin("https://example.com/dir1", "dir2")
Out[2]: 'https://example.com/dir2'  # replaced 🤔 <-- this is the one I think would cause a problem here

In [3]: urljoin("https://example.com/dir1/", "dir2")
Out[3]: 'https://example.com/dir1/dir2'  # expected

In [4]: urljoin("https://example.com/dir1/", "/dir2")
Out[4]: 'https://example.com/dir2'  # maybe expected?

In [5]: urljoin("https://example.com/dir1/", "dir2/")
Out[5]: 'https://example.com/dir1/dir2/'  # expected
```

so this change is OK iff 1) the `url` always has a trailing slash, which I don't think is currently gurarnteed? and 2) the thing being joined never has a leading slash, which is true here but a potential footgun